hphp/hack/src/parser/aast_check.rs

```diff
@@ -92,7 +108,20 @@ impl Checker {
                         self.add_error(&hint.0, syntax_error::invalid_awaitable_arity)
                     }
                 }
-                None => self.add_error(&hint.0, syntax_error::invalid_async_return_hint),
+                None => {
+                  let (start_offset, end_offset) = &hint.0.info_raw(); 
+                  let edits = vec![
+                      (*start_offset, *start_offset, "Awaitable<".into()),
+                      (*end_offset, *end_offset, ">".into())
+                  ];
+                  self.add_error_with_quickfix(
+                    *start_offset,
+                    *end_offset,
```

Comment:
You're taking a reference to this tuple, then immediately dereferencing it. You can probably remove the `&` reference, and then the `*` dereferencing shouldn't be necessary.

```suggestion
                  let (start_offset, end_offset) = hint.0.info_raw(); 
                  let edits = vec![
                      (start_offset, start_offset, "Awaitable<".into()),
                      (end_offset, end_offset, ">".into())
                  ];
                  self.add_error_with_quickfix(
                    start_offset,
                    end_offset,
```